### PR TITLE
Enable chat clickable weblinks by default

### PR DESCRIPTION
**Pull request by Froggo8311**
Make the ability to click on chat weblinks enabled by default. This does not relate to the roadmap, and is instead a "quality of life" commit.

## How to test
* Build minetest, and don't modify the config file
* Start a game
* Type a weblink into chat
* It should be enabled by default

### CODE REVIEW

**Comment by rollerozxa** [comment]
+1 for saner defaults!

**Comment by rubenwardy** [comment]
There's almost definitely a reason why this wasn't enabled by default, perhaps issues with the implementation. I don't necessarily think this should be a setting at all, once the feature is stabilised

**Comment by pecksin** [comment]
For the record, the main reason it was not enabled by default is it is useless on touchscreens, and I didn't know if it could cause complications/performance issues in that environment.

There's no specific reason I'm aware of that it would, but I know nothing about android.

**Comment by Desour** [comment]
Without having to change the setting, the user will not know that he has to middle-click or ctrl-left-click. This might lead to quite some confusion.

**Comment by Zughy** [comment]
:+1: 
Question: why chat weblinks should be disabled in the first place? Or at least, why should the client decide? I can picture a situation where a server admin doesn't want that to avoid spam (even if I can still put the link and people can manually copy it, so it would only work with very long links), but clients?

**Comment by debagos** [comment]
I think it isn't a bad thing having a way to disable the coloring of web links.
Please keep the ability.

File: src/defaultsettings.cpp [hunk 1]
```diff
@@ -65,7 +65,7 @@ void set_default_settings()
 	settings->setDefault("max_out_chat_queue_size", "20");
 	settings->setDefault("pause_on_lost_focus", "false");
 	settings->setDefault("enable_register_confirmation", "true");
-	settings->setDefault("clickable_chat_weblinks", "false");
```
Comment (rubenwardy) [inline src/defaultsettings.cpp:68]: This should be disabled on Android/ENABLE_TOUCHSCREENS still

File: minetest.conf.example [hunk 2]
```diff
@@ -1107,7 +1107,7 @@
 
 #    Clickable weblinks (middle-click or Ctrl+left-click) enabled in chat console output.
 #    type: bool
-# clickable_chat_weblinks = false
```
Comment (rubenwardy) [inline minetest.conf.example:1110]: Need to update builtin/settingtypes.txt

Comment (Froggo8311) [reply minetest.conf.example:1179]: [This commit](https://github.com/Froggo8311/minetest/commit/72fc1acfe52b45386446b66439059228117fb38d) should take care of that.

File: src/defaultsettings.cpp (hunk 1 above)
Comment (Froggo8311) [reply src/defaultsettings.cpp:68]: How would I impliment such a configuration? Thanks.

Comment (rollerozxa) [reply src/defaultsettings.cpp:68]: @Froggo8311 something like this should work:
```cpp
#ifndef HAVE_TOUCHSCREENGUI
	settings->setDefault("clickable_chat_weblinks", "true");
#else
	settings->setDefault("clickable_chat_weblinks", "false");
#endif
```

Comment (Froggo8311) [reply src/defaultsettings.cpp:68]: @rollerozxa Ok, thanks.

Should be right in [this commit](https://github.com/Froggo8311/minetest/commit/943239d19bc71f69a83f4324c9cf790177ad3b9f) now.

File: src/defaultsettings.cpp [hunk 3]
```diff
@@ -65,7 +65,7 @@ void set_default_settings()
 	settings->setDefault("max_out_chat_queue_size", "20");
 	settings->setDefault("pause_on_lost_focus", "false");
 	settings->setDefault("enable_register_confirmation", "true");
-	settings->setDefault("clickable_chat_weblinks", "true");
+	// settings->setDefault("clickable_chat_weblinks", "true"); // Moved below
```
Comment (rubenwardy) [inline src/defaultsettings.cpp:68]: there's no need to do this

Comment (Froggo8311) [reply src/defaultsettings.cpp:68]: Ok

**Review by rubenwardy — APPROVED** [review]

**Review by sfan5 — APPROVED** [review]

**Pull request merged**